templates/todo/common/auth/py/AzureDeveloperCliCredential.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from datetime import datetime
+import json
+import os
+import platform
+import re
+import subprocess
+import sys
+import time
+from typing import Any, List
+
+import six
+
+from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
+
+from azure.identity import CredentialUnavailableError
+from azure.identity._internal.decorators import log_get_token
+
+CLI_NOT_FOUND = "Azure Developer CLI could not be found. Please visit https://aka.ms/azure-dev for installation instructions and then, once installed, authenticate to your Azure account using 'azd login'."
+COMMAND_LINE = "azd auth token --output json --scope {}"
+NOT_LOGGED_IN = "Please run 'azd login' from a command prompt to authenticate before using this credential."
+
+
+class AzureDeveloperCliCredential(object):
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def close(self) -> None:
+        """Calling this method is unnecessary."""
+
+    @log_get_token("AzureDeveloperCliCredential")
+    def get_token(self, *scopes: str, **kwargs) -> AccessToken:
+        commandString = ' --scope '.join(scopes)
+        command = COMMAND_LINE.format(commandString)
+        output = _run_command(command)
+        token = parse_token(output)
+        if not token:
+            sanitized_output = sanitize_output(output)
+            raise ClientAuthenticationError(
+                message="Unexpected output from Azure Developer CLI: '{}'. \n".format(sanitized_output))
+
+        return token
+
+
+def parse_token(output):
+    try:
+        token = json.loads(output)
+        dt = datetime.strptime(token["expiresOn"], "%Y-%m-%dT%H:%M:%SZ")
+        if hasattr(dt, "timestamp"):
+            # Python >= 3.3
+            expires_on = dt.timestamp()
+        else:
+            # taken from Python 3.5's datetime.timestamp()
+            expires_on = time.mktime((dt.year, dt.month, dt.day, dt.hour, dt.minute, dt.second, -1, -1, -1))
```

Comment:
```suggestion
        expires_on = dt.timestamp()
```

Azd has a dependency on Python version 3.7:

![image](https://user-images.githubusercontent.com/24213737/205760580-08b1e055-fe72-48b2-b9f8-ac96036f03e6.png)

So, we don't need this condition to support python <= 3.3